### PR TITLE
chore: migrate zaakafhandelparameterbeheerservice to kotlin

### PR DESCRIPTION
**Pull request by edgarvonk**
Migrated zaakafhandelparameterbeheerservice to Kotlin.

Solves PZ-3700

### CODE REVIEW

File: src/main/kotlin/net/atos/zac/admin/ZaakafhandelParameterBeheerService.kt [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+        )
+        query.select(root).where(builder.equal(root.get<Any>(ZaakafhandelParameters.ZAAKTYPE_UUID), zaaktypeUUID))
+        val resultList = entityManager.createQuery(query).resultList
+        if (resultList.isNotEmpty()) {
+            return resultList.first()
+        } else {
+            return ZaakafhandelParameters().apply {
+                zaakTypeUUID = zaaktypeUUID
+            }
+        }
```
Comment (hsiliev) [inline src/main/kotlin/net/atos/zac/admin/ZaakafhandelParameterBeheerService.kt:63]:
```suggestion
        if (resultList.isNotEmpty()) {
            return resultList.first()
        }
        return ZaakafhandelParameters().apply {
            zaakTypeUUID = zaaktypeUUID
        }
```

File: src/main/kotlin/net/atos/zac/admin/ZaakafhandelParameterBeheerService.kt [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+        return entityManager.createQuery(query).resultList
+    }
+
+    fun createZaakafhandelParameters(zaakafhandelParameters: ZaakafhandelParameters): ZaakafhandelParameters {
+        zaakafhandelParameterService.clearListCache()
+        ValidationUtil.valideerObject(zaakafhandelParameters)
+        zaakafhandelParameters.humanTaskParametersCollection.forEach { ValidationUtil.valideerObject(it) }
+        zaakafhandelParameters.userEventListenerParametersCollection.forEach { ValidationUtil.valideerObject(it) }
+        zaakafhandelParameters.mailtemplateKoppelingen.forEach { ValidationUtil.valideerObject(it) }
+        zaakafhandelParameters.creatiedatum = ZonedDateTime.now()
```
Comment (hsiliev) [inline src/main/kotlin/net/atos/zac/admin/ZaakafhandelParameterBeheerService.kt:79]: We can probably use `let`, `with` here

**Review by hsiliev — APPROVED** [review]

File: src/main/kotlin/net/atos/zac/admin/ZaakafhandelParameterBeheerService.kt (hunk 1 above)
Comment (edgarvonk) [reply src/main/kotlin/net/atos/zac/admin/ZaakafhandelParameterBeheerService.kt:63]: Made it into:

```return if (resultList.isNotEmpty()) {
            resultList.first()
        } else {
            ZaakafhandelParameters().apply {
                zaakTypeUUID = zaaktypeUUID
            }
        }``` :-)

File: src/main/kotlin/net/atos/zac/admin/ZaakafhandelParameterBeheerService.kt (hunk 2 above)
Comment (edgarvonk) [reply src/main/kotlin/net/atos/zac/admin/ZaakafhandelParameterBeheerService.kt:79]: I simplified this but in a different way. ;-)

**Pull request merged**